src/snowflake/connector/cursor.py

```diff
@@ -141,6 +142,12 @@ def exit_handler(*_):  # pragma: no cover
     sys.exit(1)
 
 
+class ResultState(Enum):
+    DEFAULT = 1
+    VALID = 2
+    RESET = 3
```

Comment:
Three states just to handle the reset behaviour gracefully. When we decided this is not needed, we can remove the ResultState and any variables/conditions that depend on it.